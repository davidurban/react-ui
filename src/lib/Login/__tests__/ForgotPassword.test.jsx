import React from 'react';
import {
  shallow,
  mount,
} from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';
import { ForgotPassword } from '../index';

jest.mock('../../Icon/load-material-design-icons');

describe('rendering', () => {
  it('renders correctly', () => {
    const tree = shallow(<ForgotPassword />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly with all props', () => {
    const tree = shallow(<ForgotPassword
      logo="http://satyr.io/100x100/33?text=logo"
      title="Company"
      error="Error message"
      footer={(
        <a href="http://example.com">link</a>
      )}
    />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});

describe('functionality', () => {
  it('calls submitHandler() on submit', () => {
    const spy = sinon.spy();
    const component = mount(<ForgotPassword submitHandler={spy} />);

    component
      .find('form')
      .simulate('submit');
    expect(spy.calledOnce).toEqual(true);
  });
});
