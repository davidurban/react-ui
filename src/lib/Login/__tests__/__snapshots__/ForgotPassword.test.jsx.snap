// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly 1`] = `
<WithTranslationContextComponent
  translations={null}
>
  <ForgotPassword
    error={null}
    footer={null}
    logoUrl={null}
    onChangeHandler={null}
    submitHandler={null}
    title={null}
    translations={
      Object {
        "email": "E-mail",
        "resetPassword": "Reset password",
      }
    }
  >
    <div
      className="root"
    >
      <div
        className="box"
      >
        <form
          onSubmit={[Function]}
        >
          <TextField
            changeHandler={[Function]}
            description={null}
            disabled={false}
            error={null}
            fieldId="resetEmail"
            isLabelVisible={true}
            label="E-mail"
            placeholder={null}
            required={true}
            type="email"
          >
            <div
              className="root"
            >
              <label
                htmlFor="resetEmail"
              >
                <div
                  className="isLabelRequired"
                >
                  E-mail
                </div>
                <input
                  className="input"
                  disabled={false}
                  id="resetEmail"
                  onChange={[Function]}
                  placeholder={null}
                  required={true}
                  type="email"
                />
              </label>
            </div>
          </TextField>
          <Button
            block={true}
            clickHandler={null}
            disabled={false}
            icon={null}
            iconPosition="before"
            isLabelVisible={true}
            label="Reset password"
            loading={false}
            priority="primary"
            size="medium"
            type="submit"
            variant="default"
          >
            <button
              className="root
                    priorityPrimary
                    sizeMedium
                    
                    
                    block"
              disabled={false}
              onClick={null}
              title={null}
              type="submit"
            >
              <span>
                Reset password
              </span>
            </button>
          </Button>
        </form>
      </div>
    </div>
  </ForgotPassword>
</WithTranslationContextComponent>
`;

exports[`rendering renders correctly 2`] = `
<WithTranslationContextComponent
  translations={
    Object {
      "email": "E-mail",
      "resetPassword": "Reset password",
    }
  }
>
  <ForgotPassword
    error={null}
    footer={null}
    logoUrl={null}
    onChangeHandler={null}
    submitHandler={null}
    title={null}
    translations={
      Object {
        "email": "E-mail",
        "resetPassword": "Reset password",
      }
    }
  >
    <div
      className="root"
    >
      <div
        className="box"
      >
        <form
          onSubmit={[Function]}
        >
          <TextField
            changeHandler={[Function]}
            description={null}
            disabled={false}
            error={null}
            fieldId="resetEmail"
            isLabelVisible={true}
            label="E-mail"
            placeholder={null}
            required={true}
            type="email"
          >
            <div
              className="root"
            >
              <label
                htmlFor="resetEmail"
              >
                <div
                  className="isLabelRequired"
                >
                  E-mail
                </div>
                <input
                  className="input"
                  disabled={false}
                  id="resetEmail"
                  onChange={[Function]}
                  placeholder={null}
                  required={true}
                  type="email"
                />
              </label>
            </div>
          </TextField>
          <Button
            block={true}
            clickHandler={null}
            disabled={false}
            icon={null}
            iconPosition="before"
            isLabelVisible={true}
            label="Reset password"
            loading={false}
            priority="primary"
            size="medium"
            type="submit"
            variant="default"
          >
            <button
              className="root
                    priorityPrimary
                    sizeMedium
                    
                    
                    block"
              disabled={false}
              onClick={null}
              title={null}
              type="submit"
            >
              <span>
                Reset password
              </span>
            </button>
          </Button>
        </form>
      </div>
    </div>
  </ForgotPassword>
</WithTranslationContextComponent>
`;

exports[`rendering renders correctly with all props except translations 1`] = `
<WithTranslationContextComponent
  error="Error message"
  footer={
    <a
      href="http://example.com"
    >
      link
    </a>
  }
  logo="http://satyr.io/100x100/33?text=logo"
  title="Company"
  translations={null}
>
  <ForgotPassword
    error="Error message"
    footer={
      <a
        href="http://example.com"
      >
        link
      </a>
    }
    logo="http://satyr.io/100x100/33?text=logo"
    logoUrl={null}
    onChangeHandler={null}
    submitHandler={null}
    title="Company"
    translations={
      Object {
        "email": "E-mail",
        "resetPassword": "Reset password",
      }
    }
  >
    <div
      className="root"
    >
      <div
        className="title"
      >
        Company
      </div>
      <div
        className="box"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="error"
          >
            Error message
          </div>
          <TextField
            changeHandler={[Function]}
            description={null}
            disabled={false}
            error={null}
            fieldId="resetEmail"
            isLabelVisible={true}
            label="E-mail"
            placeholder={null}
            required={true}
            type="email"
          >
            <div
              className="root"
            >
              <label
                htmlFor="resetEmail"
              >
                <div
                  className="isLabelRequired"
                >
                  E-mail
                </div>
                <input
                  className="input"
                  disabled={false}
                  id="resetEmail"
                  onChange={[Function]}
                  placeholder={null}
                  required={true}
                  type="email"
                />
              </label>
            </div>
          </TextField>
          <Button
            block={true}
            clickHandler={null}
            disabled={false}
            icon={null}
            iconPosition="before"
            isLabelVisible={true}
            label="Reset password"
            loading={false}
            priority="primary"
            size="medium"
            type="submit"
            variant="default"
          >
            <button
              className="root
                    priorityPrimary
                    sizeMedium
                    
                    
                    block"
              disabled={false}
              onClick={null}
              title={null}
              type="submit"
            >
              <span>
                Reset password
              </span>
            </button>
          </Button>
        </form>
        <div
          className="footer"
        >
          <a
            href="http://example.com"
          >
            link
          </a>
        </div>
      </div>
    </div>
  </ForgotPassword>
</WithTranslationContextComponent>
`;
